tests/zquantum/core/cost_function_test.py

```diff
@@ -25,223 +26,243 @@
-):
-    params = np.array([1.0], dtype=float)
-    value = ground_state_cost_function(params)
-    assert -1 <= value <= 1
-
-
-def test_noisy_ground_state_cost_function_adds_noise_to_parameters():
-    target_operator = QubitOperator("Z0")
-    parametrized_circuit = MockAnsatz(
-        number_of_layers=2, problem_size=1
-    ).parametrized_circuit
-    parametrized_circuit.bind = mock.Mock(wraps=parametrized_circuit.bind)
-    backend = SymbolicSimulator()
-    estimation_method = estimate_expectation_values_by_averaging
-    estimation_preprocessors = [partial(allocate_shots_uniformly, number_of_shots=1)]
-    noisy_ground_state_cost_function = get_ground_state_cost_function(
-        target_operator,
-        parametrized_circuit,
-        backend,
-        estimation_method=estimation_method,
-        estimation_preprocessors=estimation_preprocessors,
-        parameter_precision=1e-4,
-        parameter_precision_seed=RNGSEED,
+BACKEND = SymbolicSimulator()
+ESTIMATION_METHOD = estimate_expectation_values_by_averaging
+ESTIMATION_PREPROCESSORS = [partial(allocate_shots_uniformly, number_of_shots=1)]
+
+
+class TestGroundStateCostFunction:
+    @pytest.fixture(
+        params=[
+            {
+                "target_operator": QubitOperator("Z0"),
+                "parametrized_circuit": MockAnsatz(
+                    number_of_layers=1, problem_size=1
+                ).parametrized_circuit,
+            },
+            {
+                "target_operator": QubitOperator("Z0 Z1"),
+                "parametrized_circuit": MockAnsatz(
+                    number_of_layers=1, problem_size=2
+                ).parametrized_circuit,
+            },
+            {
+                "target_operator": QubitOperator("Z0 Z1"),
+                "parametrized_circuit": MockAnsatz(
+                    number_of_layers=2, problem_size=2
+                ).parametrized_circuit,
+                "fixed_parameters": [1.2],
+            },
+            {
+                "target_operator": QubitOperator("Z0 Z1"),
+                "parametrized_circuit": MockAnsatz(
+                    number_of_layers=2, problem_size=2
+                ).parametrized_circuit,
+                "fixed_parameters": [1.2],
+                "parameter_precision": 0.001,
+                "parameter_precision_seed": RNGSEED,
+            },
+            {
+                "target_operator": QubitOperator("Z0"),
+                "parametrized_circuit": MockAnsatz(
+                    number_of_layers=1, problem_size=1
+                ).parametrized_circuit,
+            },
+        ]
     )
+    def ground_state_cost_function(self, request):
+        return get_ground_state_cost_function(
+            **request.param,
+            backend=BACKEND,
+            estimation_method=ESTIMATION_METHOD,
+            estimation_preprocessors=ESTIMATION_PREPROCESSORS
+        )
 
-    generator = np.random.default_rng(RNGSEED)
+    def test_returns_value_between_plus_and_minus_one(
```

Comment:
For this one I would do pytest.mark.parameterize and iterate over more potential parameter values than the one provided.